### PR TITLE
Adding a few serialization attributes back to the reference assemblies

### LINKED ISSUE #35561: Should GenAPI be removing the XML Serialization attributes?
Currently the automatic generation of reference assemblies excludes some of the XML Serialization attributes

https://github.com/dotnet/corefx/blob/26b932e3c56702382d13ece010b300d3181d2891/eng/DefaultGenApiDocIds.txt#L44-L47

This feels like (but is currently unverified) it would create a problem for `sgen.exe`; in that the `sgen`-generated serializer/deserializer would behave differently than the runtime-generated serialization assembly; so it seems like they should be left as part of the ref.cs

Alternatively, it seems like all of the attributes in https://docs.microsoft.com/en-us/dotnet/standard/serialization/attributes-that-control-xml-serialization?view=netframework-4.7.2 would be removed.

No matter what, the inconsistency seems weird.

cc: @ericstj

### PR DESCRIPTION
**Pull request by joperezr**
Fixes #35561 

cc: @Lxiamail @mconnew  PTAL

### CODE REVIEW

**Comment by jkotas** [comment]
Are we actually going to make sgen.exe work against reference assemblies? Or is it this just a speculative change that may be helps one day. If it is the later, we should not be adding these.

File: src/System.ServiceModel.Syndication/ref/System.ServiceModel.Syndication.csproj [hunk 1]
```diff
@@ -6,6 +6,15 @@
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="System.ServiceModel.Syndication.cs" />
+    <Compile Include="..\src\System\ServiceModel\Syndication\Atom10Constants.cs">
```
Comment (jkotas) [inline src/System.ServiceModel.Syndication/ref/System.ServiceModel.Syndication.csproj]: We do not include sources into the reference assembly. Instead, regenerate the reference assembly using `/t:GenerateReferenceSource`

Comment (ericstj) [reply src/System.ServiceModel.Syndication/ref/System.ServiceModel.Syndication.csproj]: Yeah, here is a place where we actually want duplication.  We don't want the source to change and automatically update the ref.  We need people to stop and think about it when their change impacts the reference assembly source.

**Comment by ericstj** [comment]
@mconnew can you confirm that this will be a safe change to make and won't cause version-to-version serialization issues (or if it does, prioritize the break between .NETCore versions vs break with desktop).

**Comment by joperezr** [comment]
Setting no merge until next version of master is merged into release/3.0 branch.

File: src/System.ServiceModel.Syndication/ref/System.ServiceModel.Syndication.csproj (hunk 1 above)
Comment (joperezr) [reply src/System.ServiceModel.Syndication/ref/System.ServiceModel.Syndication.csproj]: Will change.

**Comment by mconnew** [comment]
@ericstj, did we release a version which didn't have these attributes? With the syndication classes, the XmlSerializer is instantiated by the Syndication implementation itself and will do the check at runtime. As a customer could create a pre-generated serializer (which uses the ref assemblies) for these classes the attributes do need to be there otherwise the serialization would not work. Realistically it's most likely a theoretical problem as the only use case for doing so would be extremely contrived. The only thing I can think of which might be a real problem is with the UWP toolchain pregenerating serializers. I don't know if that runs against the ref assemblies or the runtime assemblies. I would presume runtime because of the decisions made about the Serializable attribute.  
I do have a couple of questions about other items in DefaultGenApiDocIds.txt though. My understanding is that if any of these attributes exist in the implementation, then it suppresses them when creating the reference assembly source. If that's correct, isn't it a mistake to remove System.ComponentModel.EditorBrowsableAttribute and System.ObsoleteAttribute? VS uses the reference assemblies for intellisense so needs the EditorBrowsableAttribute to exist in the reference assembly to suppress intellisense. And the Obsolete attribute is used by Roslyn to throw a warning or an error at compile time and Roslyn only uses the reference assemblies too. And without the DefaultMemberAttribute in the reference assembly, Roslyn would emit the wrong code if a class used anything other than `this` as the default indexer property. If the policy is that you corefx libraries never specifies a different default indexer property, do we have a code analysis rule to enforce this?

**Comment by joperezr** [comment]
Preview 5 has snapped so this is now ready to go

**Comment by ericstj** [comment]
> I do have a couple of questions about other items

There are issues tracking the other omissions.  In general this list just means that they won't be validated by APICompat and if a dev tried to regenerate source they'd be omitted, but it doesn't mean they are missing (since source for reference assemblies is checked in).  Part of the goal of these issues is to get the tools/source in a better state where we can auto-generate reference source.

> did we release a version which didn't have these attributes

Yes, you released 4.5.0 without those attributes. https://www.nuget.org/packages/System.ServiceModel.Syndication/4.5.0

**Comment by joperezr** [comment]
> Yes, you released 4.5.0 without those attributes. https://www.nuget.org/packages/System.ServiceModel.Syndication/4.5.0

@mconnew if this is a concern, I would suggest for your team to take over this issue and instead of merging this PR do the required changes in your side so that this is not making matters worse.

**Comment by joperezr** [comment]
I've reset your review status @mconnew, if after @ericstj answer you still feel comfortable with this and sign off, then I'll go ahead and merge.

**Comment by mconnew** [comment]
@joperezr, the current PR is correct. Having them missing from the contract only affects a really obscure use case which I suspect nobody on the planet actually does even on .NET Framwork. While 4.5.0 is technically broken, it's not a concern in practicality. So the right thing is to make sure they are there in the next release.

**Comment by joperezr** [comment]
@mconnew thanks for confirming 😄

**Pull request merged**